lib/roken/getauxval.c

```diff
@@ -213,6 +213,7 @@ rk_getauxval(unsigned long type)
 
     getauxval_sets_errno = 0;
     errno = save_errno;
+    return 0;
```

Comment:
Reading the getauxval() the correct return value here should be "ret2" which is the value successfully returned by the getauxval() call on line 207.

Reply:
your choices are 0 or ret. ret2 is a throwaway value. i think ret (not ret2) is correct

Reply:
actually, look at the code above this. shouldn't it
   
    if ((a = rk_getauxv(type)) == NULL) {
        errno = ENOENT;
        return 0;
    }
    return a->a_un.a_val;